### PR TITLE
Make schedulerSize of PartitionPumpManager configurable

### LINKED ISSUE #22942: [FEATURE REQ] Make schedulerSize of PartitionPumpManager configurable
**Is your feature request related to a problem? Please describe.**
The scheduler thread pool of the `PartitionPumpManager` that works behind a `EventProcessorClient` in `azure-messaging-eventhubs` always uses 4 * the number of available CPU cores as its size. There are situations, where this default is not appropriate. For example, we have a service that processes events from multiple Event Hubs and uses multiple `EventProcessorClient`s. Thus, theres a large number of threads contending for CPU time. If we were able to reduce the size of each thread pool, there would be less overhead lost by context switching.

**Describe the solution you'd like**
I'd like to have a way to explicitly set the number of threads used by the scheduler that works behind the `PartitionPumpManager`.

**Describe alternatives you've considered**
[`-XX:ActiveProcessorCount`](https://docs.oracle.com/en/java/javase/11/tools/java.html#GUID-3B1CE181-CD30-4178-9602-230B800D4FAE__GUID-6155D02B-FD3F-47BF-BD06-E3D111919353) can be used to limit the number of processors used to calculate the scheduler size, but this will affect the entire application and might not be desirable.

**Additional context**
This is the piece of code that sets the default, which I'd like to change: https://github.com/Azure/azure-sdk-for-java/blob/azure-messaging-eventhubs_5.9.0/sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java#L65-L67

### PR DESCRIPTION
**Pull request by liukun-msft**
# Description

- Fixes #22942

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/PartitionPumpOptions.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.models;
+
+import java.util.Objects;
+
+/**
+ * Set of options that can be specified when build each partition pump.
+ */
+public class PartitionPumpOptions {
```
Comment (liukun-msft) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/PartitionPumpOptions.java:11]: Add PartitionPumpOptions to let user configure partition pumps.  And pass this to PartitionPumpManager.
Currently the configuration is applied to all partition pumps. Should we need have ability to configure for each partition?

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java [hunk 2]
```diff
@@ -221,8 +224,8 @@ void startPartitionPump(PartitionOwnership claimedOwnership, Checkpoint checkpoi
             ReceiveOptions receiveOptions = new ReceiveOptions().setOwnerLevel(0L)
                 .setTrackLastEnqueuedEventProperties(trackLastEnqueuedEventProperties);
 
-            Scheduler scheduler = Schedulers.newBoundedElastic(schedulerSize,
-                MAXIMUM_QUEUE_SIZE, "partition-pump-" + claimedOwnership.getPartitionId());
+            Scheduler scheduler = Schedulers.newBoundedElastic(this.partitionPumpOptions.getSchedulerSize(),
+                this.partitionPumpOptions.getMaxQueueSize(), "partition-pump-" + claimedOwnership.getPartitionId());
```
[the remark this answers is not in the document]
Comment (anuchandy) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java:228]: If more and more customers ask and this becomes a priority, we could measure it i.e., record resource utilization in aks-stress-infra/VM.

Each category runs under the same physical resource cap cpu+memory and does an IO operation, e.g., sends an event to EH after each batch. So two sets - the first set with default BEScheduler config and the second set with parameterized BEScheduler configured according to the category. This way, we could get to compare resource utilization and throughput (ops/sec) for both sets.

[the remark this answers is not in the document]
Comment (liukun-msft) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java:228]: Thanks for comments! Yes, you are right. We need data to prove whether this feature is valid or not.
I will move to set up the stress test framework, which can benefit for this feature.

[the remark this answers is not in the document]
Comment (anuchandy) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java:228]: Sounds good, thanks Kun!

**Pull request closed without merging**